src/sorting_topic.py

```diff
@@ -8,3 +8,35 @@ def selection_sort(collection):
         collection.remove(current_smallest)
         sorted_collection.append(current_smallest)
     return sorted_collection
+
+
+def insertion_sort(collection):
+    sorted_collection = []
```

Comment:
Insertion sort is typically in-place. There's no need to make a copy of the entire collection.

Really, same goes for all sorts. Either sort in-place or return a sorted copy while leaving the original untouched.

Reply:
_Not_ in place was intentional as I feel having a new list will be more accessible at this stage. Bubble sort does not really lend itself to not in-place, but good call on having it return a copy to have no side effect.